server/controllers/admin/employees.js

```diff
@@ -190,43 +216,43 @@ exports.update = function update(req, res, next) {
     employee.date_embauche = new Date(employee.date_embauche);
   }
 
-  db.exec(sql, [employee, req.params.id])
-  .then(function (row) {
+  delete(employee.debitor_group_uuid);
+  delete(employee.creditor_group_uuid);
+  delete(employee.text);
+  delete(employee.basic_salary);
+  delete(employee.fonction_txt);
+  delete(employee.code_grade);
+  delete(employee.debitor_text);
+  delete(employee.creditor_text);
+  delete(employee.creditor_text);
+  delete(employee.account_id);
+  delete(employee.code_employee);
+
+
+  var updateCreditor = 'UPDATE creditor SET ? WHERE creditor.uuid = ?';
+  var updateDebitor = 'UPDATE debitor SET ? WHERE debitor.uuid = ?';
+  var sql = 'UPDATE employee SET ? WHERE employee.id = ?';
 
-    if (!row.affectedRows) {
-      throw new req.codes.ERR_NOT_FOUND();
-    }
+  transaction = db.transaction();
 
-    var sql2 =
-      'SELECT employee.id, employee.code AS code_employee, employee.prenom, employee.name, ' +
-        'employee.postnom, employee.sexe, employee.dob, employee.date_embauche, employee.service_id, ' +
-        'employee.nb_spouse, employee.nb_enfant, employee.grade_id, employee.locked, grade.text, grade.basic_salary, ' +
-        'fonction.id AS fonction_id, fonction.fonction_txt, service.name AS service_txt, ' +
-        'employee.phone, employee.email, employee.adresse, employee.bank, employee.bank_account, ' +
-        'employee.daily_salary, employee.location_id, grade.code AS code_grade, debitor.uuid as debitor_uuid, ' +
-        'debitor.text AS debitor_text,debitor.group_uuid as debitor_group_uuid, ' +
-        'creditor.uuid as creditor_uuid, creditor.text AS creditor_text, ' +
-        'creditor.group_uuid as creditor_group_uuid, creditor_group.account_id ' +
-      'FROM employee ' +
-        'JOIN grade ON employee.grade_id = grade.uuid ' +
-        'JOIN fonction ON employee.fonction_id = fonction.id ' +
-        'JOIN debitor ON employee.debitor_uuid = debitor.uuid ' +
-        'JOIN creditor ON employee.creditor_uuid = creditor.uuid ' +
-        'JOIN creditor_group ON creditor_group.uuid = creditor.group_uuid ' +
-        'LEFT JOIN service ON service.id = employee.service_id ' +
-      'WHERE employee.id = ? ';
-
-    return db.exec(sql2, [req.params.id]);
-  })
-  .then(function (rows) {
+  transaction
+    .addQuery(updateCreditor, [creditor, creditor.uuid])
+    .addQuery(updateDebitor, [debitor, debitor.uuid])
+    .addQuery(sql, [employee, req.params.id]);
 
-    if (!rows.length) {
-      throw new req.codes.ERR_NOT_FOUND();
-    }
-    res.status(200).json(rows[0]);
-  })
-  .catch(next)
-  .done();
+  transaction.execute()
+    .then(function (results) {
+      if (!results[2].affectedRows) {
```

Comment:
Ah, I see now.  It looks like you are trying to get the employee id to send back to the client. Thanks!